### PR TITLE
 Migrate S3735 to eslint based implementation

### LINKED ISSUE #2122: Migrate S3735 to eslint implementation (new rule for TS)

### PR DESCRIPTION
**Pull request by guillaume-dequenne-sonarsource**
Fixes #2122

### CODE REVIEW

File: eslint-bridge/src/rules/void-use.ts [hunk 1]
```diff
@@ -17,16 +17,23 @@
...
 
-public class VoidUseCheckTest {
+export const rule: Rule.RuleModule = {
+  create(context: Rule.RuleContext) {
+    function checkNode(node: estree.Node) {
+      const unaryExpression: estree.UnaryExpression = node as estree.UnaryExpression;
+      if (unaryExpression.argument.type === 'Literal' && 0 === unaryExpression.argument.value) {
+        return;
+      }
+      context.report({ node, message: 'Use "null" instead.' });
```
Comment (andrea-guarino-sonarsource) [inline eslint-bridge/src/rules/void-use.ts:32]: Why did you change the message?

File: eslint-bridge/tests/rules/void-use.test.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            `,
+    },
+  ],
+  invalid: [
+    {
+      code: `
+            void function() {
+            }()
+            `,
+      errors: 1,
```
Comment (andrea-guarino-sonarsource) [inline eslint-bridge/tests/rules/void-use.test.ts:51]: Please add assertion for primary location

Comment (guillaume-dequenne-sonarsource) [reply eslint-bridge/tests/rules/void-use.test.ts]: Did that, and actually updated the way we report to better reflect our initial implementation.

File: eslint-bridge/src/rules/void-use.ts (hunk 1 above)
Comment (guillaume-dequenne-sonarsource) [reply eslint-bridge/src/rules/void-use.ts]: That was not intended, not even sure where I found this one 😄

**Review by andrea-guarino-sonarsource — APPROVED** [review]
LGTM
